src/main/java/Duke.java

```diff
@@ -1,10 +1,159 @@
+import java.io.*;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
 public class Duke {
+    public static String h_rule = "────────────────────────────────────────────────────────────────────";
+
+    public static int dukeRunner(String log) throws DukeNotFoundException, DukeDescriptionException, DukeTimingException, FileNotFoundException {
+        Scanner sc = new Scanner(System.in);
+
+        String raw_in;
+        String[] input;
+        List<Task> itemList = new ArrayList<>();
+        int count = 0;
+        List<String> fileContents = null;
+
+        try {
+            fileContents = new ArrayList<>(Files.readAllLines(Paths.get(log)));
+        } catch (IOException e) {
+            System.out.println(e);
+        }
+        while (true) {
+            raw_in = sc.nextLine().trim();
+            input = raw_in.split(" ");
+            System.out.println(h_rule);
+            String joined = "";
+            String timing = "";
+            if (raw_in.equals("bye")) {
+                System.out.println("Goodbye! Have a nice day!\n");
+                break;
+            } else {
+                switch (input[0]) {
+                    case "list":
+                        int index = 1;
+                        for (Task s : itemList) {
```

Comment:
Is there a particular reason you are using the enhanced for loop here? I see that you have an index variable, maybe a normal for loop would then suffice, as the iterator is contained within the loop's scope